matrix-synapse-riot/terraform/matrix-server/bootstrap-post-dns.sh

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env bash
+
+RIOT_VERSION=0.17.7
+
+DOMAIN_NAME=$1
+DO_TOKEN=$2
+
+DEHYDRATED_VERSION=0.6.2
+
+#######################
+# nginx + letsencrypt #
+#######################
+systemctl stop nginx.service
+
+# Generate dhparam.pem
+echo "Generating DH parameters, 4096 bit long safe prime, generator 2"
+echo "This is going to take a long time"
+openssl dhparam -out /etc/ssl/certs/dhparam.pem 4096 2> /dev/null
+
+# Copy nginx config to directory 
+systemctl restart nginx.service
+cd /etc/nginx/
+mkdir -m 700 ssl
+cp /tmp/matrix-server/nginx-chat /etc/nginx/sites-available/chat.$DOMAIN_NAME
+sed -i -e "s/CHATDNS/$DOMAIN_NAME/g" /etc/nginx/sites-available/chat.$DOMAIN_NAME
+cp /tmp/matrix-server/nginx-matrix /etc/nginx/sites-available/matrix.$DOMAIN_NAME
+sed -i -e "s/CHATDNS/$DOMAIN_NAME/g" /etc/nginx/sites-available/matrix.$DOMAIN_NAME
+ln -s /etc/nginx/sites-available/matrix.$DOMAIN_NAME /etc/nginx/sites-enabled/matrix.$DOMAIN_NAME
+ln -s /etc/nginx/sites-available/chat.$DOMAIN_NAME /etc/nginx/sites-enabled/chat.$DOMAIN_NAME
+
+# Install Dehydrated 
+cd /opt/
+wget https://github.com/lukas2511/dehydrated/releases/download/v$DEHYDRATED_VERSION/dehydrated-$DEHYDRATED_VERSION.tar.gz -O /opt/dehydrated-$DEHYDRATED_VERSION.tar.gz
+tar xf dehydrated-$DEHYDRATED_VERSION.tar.gz
+rm dehydrated-$DEHYDRATED_VERSION.tar.gz
+chmod 700 /opt/dehydrated-$DEHYDRATED_VERSION
+cd /opt/dehydrated-$DEHYDRATED_VERSION
+cp /tmp/matrix-server/dehydrated-config /opt/dehydrated-$DEHYDRATED_VERSION/config
+cp /tmp/matrix-server/dehydrated-hooks /opt/dehydrated-$DEHYDRATED_VERSION/hooks.sh
+echo -n "chat.$DOMAIN_NAME matrix.$DOMAIN_NAME" > /opt/dehydrated-$DEHYDRATED_VERSION/domains.txt
+chmod +x /tmp/matrix-server/dehydrated-hooks /opt/dehydrated-$DEHYDRATED_VERSION/hooks.sh
+ln -s /opt/dehydrated-$DEHYDRATED_VERSION /opt/dehydrated
+cd /opt/dehydrated
+sed -i -e "s/TOKENSTRING/$DO_TOKEN/g" hooks.sh
+sed -i -e "s/DODOMAIN/$DOMAIN_NAME/g" hooks.sh
+
+# Register to Let's Encrypt
+./dehydrated --register --accept-terms
+
+cp /tmp/matrix-server/dehydrated-renew.sh /usr/local/sbin/dehydrated-renew.sh
+
+# Configure auto-renewals
+sh -c '(crontab -l 2>/dev/null; echo "@weekly /usr/local/sbin/dehydrated-renew.sh") | crontab -'
+
+############
+# iptables #
+############
+cp /tmp/matrix-server/ip6tables-rules /etc/iptables/rules.v6
+cp /tmp/matrix-server/iptables-rules /etc/iptables/rules.v4
```

Comment:
Don't you need to:

```
iptables-apply rules.v4
ip6tables-apply rules.v6
```

To apply the rules in these files?

Reply:
Added:
```
ip6tables-restore /etc/iptables/rules.v6
iptables-restore /etc/iptables/rules.v4
```
instead of iptables-apply because it needs to be interactive.